File: frontend/src/app/model/survey.ts
import { User } from "./user";

export class Survey {
    lessThan50kg!: boolean;
    sicknessSymptoms!: boolean;
    skinIllness!: boolean;
    bloodPressureAbnormalities!: boolean;
    therapyIntake!: boolean;
    menstrualCycle!: boolean;
    dentalInterventions!: boolean;
    skinPiercings!: boolean;
    donor!: string;

  public constructor(obj?: any) {
      if (obj) {
          this.lessThan50kg = obj.lessThan50kg;
          this.sicknessSymptoms = obj.sicknessSymptoms;
          this.skinIllness = obj.skinIllness;
          this.bloodPressureAbnormalities = obj.anbloodPressureAbnormalitiesswer4;
          this.therapyIntake = obj.therapyIntake;
          this.menstrualCycle = obj.menstrualCycle;
          this.dentalInterventions = obj.dentalInterventions;
          this.skinPiercings = obj.skinPiercings;
          this.donor = obj.donor;
      }
  }


}

File: backend/src/main/java/group8/bloodbank/model/Survey.java
package group8.bloodbank.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Survey {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @OneToOne
    @JoinColumn(name = "donor_id")
    private Donor donor;
    @Column
    private boolean lessThan50kg;
    @Column
    private boolean sicknessSymptoms;
    @Column
    private boolean skinIllness;
    @Column
    private boolean bloodPressureAbnormalities;
    @Column
    private boolean therapyIntake;
    @Column
    private boolean menstrualCycle;
    @Column
    private boolean dentalInterventions;
    @Column
    private boolean skinPiercings;
}